runtime-modules/forum/src/lib.rs

```diff
@@ -1198,12 +1228,56 @@ impl<T: Trait> Module<T> {
         // Make sure thread exists
         let thread = Self::ensure_thread_exists(category_id, thread_id)?;
 
+        if thread.archived {
+            return Err(ERROR_THREAD_IMMUTABLE);
+        }
+
         // and corresponding category is mutable
-        Self::ensure_catgory_is_mutable(thread.category_id)?;
+        Self::ensure_category_is_mutable(thread.category_id)?;
 
         Ok(thread)
     }
 
+    fn ensure_can_update_thread_archival_status(
+        origin: T::Origin,
+        actor: &PrivilegedActor<T>,
+        category_id: &T::CategoryId,
+        thread_id: &T::ThreadId,
+    ) -> Result<
+        (
+            Category<T::CategoryId, T::ThreadId, T::Hash>,
+            Thread<T::ForumUserId, T::CategoryId, T::Moment, T::Hash>,
+        ),
+        &'static str,
+    > {
+        // Check actor's role
+        match actor {
+            PrivilegedActor::Lead => Self::ensure_is_forum_lead(origin)?,
+            PrivilegedActor::Moderator(moderator_id) => {
+                Self::ensure_is_moderator(origin, &moderator_id)?
+            }
+        };
+
+        let thread = Self::ensure_thread_is_mutable(category_id, thread_id)?;
+        let category = <CategoryById<T>>::get(category_id);
+
+        // Closure ensuring moderator can delete category
+        let ensure_moderator_can_update = |moderator_id: &T::ModeratorId| -> dispatch::Result {
+            Self::ensure_can_moderate_category_path(moderator_id, &category_id)
+                .map_err(|_| ERROR_MODERATOR_CANT_UPDATE_CATEGORY)?;
+
+            Ok(())
+        };
+
+        // Decide if actor can delete category
```

Comment:
Why is this comment talking about deletion?

Reply:
Mistake - probably a line of code copied from elsewhere. It's fixed now